.github/workflows/all-tests.yml

```diff
@@ -32,13 +32,77 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-          # cache: 'pip' # caching pip dependencies
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v41
+
+      - name: Set environment variables based on changed files
+        id: modules-changed
+        shell: bash
+        run: |
+          declare -A changes=(
+            ["pyproject.toml"]="CORE_CHANGED"
+            ["/core/"]="CORE_CHANGED"
+            ["/extractors/neoextractors/neobaseextractor.py"]="CORE_CHANGED"
+            ["/extractors/"]="EXTRACTORS_CHANGED"
+            ["plexon2"]="PLEXON2_CHANGED"
+            ["/preprocessing/"]="PREPROCESSING_CHANGED"
+            ["/postprocessing/"]="POSTPROCESSING_CHANGED"
+            ["/qualitymetrics/"]="QUALITYMETRICS_CHANGED"
+            ["/sorters/"]="SORTERS_CHANGED"
+            ["/sorters/external"]="SORTERS_EXTERNAL_CHANGED"
+            ["/sorters/internal"]="SORTERS_INTERNAL_CHANGED"
+            ["/comparison/"]="COMPARISON_CHANGED"
+            ["/curation/"]="CURATION_CHANGED"
+            ["/widgets/"]="WIDGETS_CHANGED"
+            ["/exporters/"]="EXPORTERS_CHANGED"
+            ["/sortingcomponents/"]="SORTINGCOMPONENTS_CHANGED"
+            ["/generation/"]="GENERATION_CHANGED"
+          )
+
+          changed_files="${{ steps.changed-files.outputs.all_changed_files }}"
```

Comment:
why not using the python script here?

Reply:
That was the coal it is just that I had to test something. 

Now all the logic to decide which tests to run lives on this script.